server/methods/core/cart.js

```diff
@@ -1011,5 +1011,31 @@ Meteor.methods({
     }
 
     return Collections.Cart.findOne(selector);
+  },
+
+  /**
+   * @method cart/setAnonymousUserEmail
+   * @summary sets email for anonymous users as the first checkout step
+   * if Stripe is the enabled payment method
+   * @param {Object} userId - current user's Id
+   * @param {String} email - email to set for anonymous user
+   * @return {Number} returns update result
+   */
+  "cart/setAnonymousUserEmail": function (userId, email) {
+    check(userId, String);
+    check(email, String);
+
+    const currentUser = Collections.Accounts.findOne({ _id: userId });
+    let newEmail = {};
+
+    if (!currentUser.emails.length) {
+      newEmail = {
+        address: email,
+        provides: "default",
+        verified: false
+      };
+    }
+
+    return Collections.Accounts.update(userId, { $push: { emails: newEmail } });
```

Comment:
This method of solving this problem opens the possibility of having several `Accounts` documents that all have the same default email address. We really need to solve this by attaching the email address to the `cart` and not by attaching an email address to an anonymous user account.

See how it's done in `orders/AddOrderEmail` for an example.
https://github.com/reactioncommerce/reaction/blob/kieha-stripe-anon-checkout-2832/server/methods/core/orders.js#L797-L824

As you can see in the [cart schema](https://github.com/reactioncommerce/reaction/blob/kieha-stripe-anon-checkout-2832/lib/collections/schemas/cart.js#L126) cart schema already supports orders.